manim/mobject/number_line.py

```diff
@@ -106,7 +106,7 @@ def __init__(
         # avoid mutable arguments in defaults
         if decimal_number_config is None:
             decimal_number_config = {
-                "num_decimal_places": 0,
+                "num_decimal_places": self.decimal_places_from_step(x_range),
                 "font_size": 24,
             }  # font_size does nothing
```

Comment:
Moving this after x_range is defined makes more sense conceptually since it is now relying on its input. x_range is split up into `x_min, x_max and x_step`. So it could just use the x_step value from there.

Reply:
Done